javascripts/discourse/components/user-color-palette-selector.gjs

```diff
@@ -0,0 +1,175 @@
+    // color scheme cookie set and other automatic things from core loads their
+    // preference.
+    if (!this.currentUser && this.selectedColorPaletteId) {
+      loadColorSchemeStylesheet(this.selectedColorPaletteId);
+    }
+  }
+
+  get userColorPalettes() {
+    return listColorSchemes(this.site);
+  }
+
+  get selectedColorPaletteId() {
+    if (this.currentUser) {
+      return this.session.userColorSchemeId;
+    }
+
+    return this.keyValueStore.getItem("anon-horizon-color-palette-id");
+  }
+
+  @action
+  onRegisterMenu(api) {
+    this.dMenu = api;
+  }
+
+  @action
+  async buildColorPaletteObject() {
+    // NOTE: The function refers to color schemes, but we actually
+    // refer to these as palettes now.
+
+    const availablePalettes = this.userColorPalettes
+      .map((userPalette) => {
+        return {
+          ...userPalette,
+          accent: `#${
+            userPalette.colors.find((color) => color.name === "tertiary").hex
+          }`,
+        };
+      })
+      .filter((userPalette) => {
+        return HORIZON_PALETTES.some((palette) => {
+          return userPalette.name.toLowerCase().includes(palette.toLowerCase());
+        });
+      })
+      .sort();
+
+    // Match the light scheme with the corresponding dark id based in the name
+    return (
+      availablePalettes
+        .map((palette) => {
+          if (palette.is_dark) {
+            return palette;
+          }
+
+          const normalizedLightName = palette.name.toLowerCase();
+
+          const correspondingDarkModeId = availablePalettes.find(
+            (item) =>
+              item.is_dark &&
+              normalizedLightName ===
+                item.name.toLowerCase().replace(/\s+dark$/, "")
+          )?.id;
+
+          return {
+            ...palette,
+            correspondingDarkModeId,
+          };
+        })
+        // Only want to show palettes that have corresponding light/dark modes
+        .filter((palette) => !palette.is_dark)
+    );
+  }
+
+  @action
+  paletteSelected(selectedPalette) {
+    this.dMenu.close();
+    if (this.interfaceColor.darkModeForced) {
+      loadColorSchemeStylesheet(selectedPalette.correspondingDarkModeId);
+      this.#updatePreference(selectedPalette);
+    } else {
+      loadColorSchemeStylesheet(selectedPalette.id);
+      this.#updatePreference(selectedPalette);
+    }
+    reload();
+  }
+
+  #updatePreference(selectedPalette) {
+    if (this.currentUser) {
+      updateColorSchemeCookie(selectedPalette.id);
+      updateColorSchemeCookie(selectedPalette.correspondingDarkModeId, {
+        dark: true,
+      });
+    } else {
+      this.keyValueStore.setItem(
+        "anon-horizon-color-palette-id",
+        selectedPalette.id
+      );
+      this.keyValueStore.setItem(
+        "anon-horizon-color-palette-dark-id",
+        selectedPalette.correspondingDarkModeId
+      );
+    }
+  }
+
+  <template>
+    {{#if this.userColorPalettes}}
```

Comment:
The icon might be displayed with a working menu if `userColorPalettes`is not empty and `buildColorPaletteObject` returns an empty list.

Reply:
This has been refactored per your suggestion so is no longer a concern

Reply:
This is still a problem after the refactor, I will fix